frontend/src/store/playbackManager.ts

```diff
@@ -1124,6 +1192,34 @@ class PlaybackManagerStore {
       }
     );
 
+    watch(
+      () => this.currentAudioStreamIndex,
+      (newVal, oldVal) => {
+        if (oldVal !== undefined && newVal !== undefined && oldVal !== newVal) {
+          this.fetchCurrentMediaSource();
+        }
+      }
+    );
+
+    watch(
+      () => ({
+        currentSubtitleStreamIndex: this.currentSubtitleStreamIndex,
+        currentSubtitleTrack: this.currentSubtitleTrack
+      }),
+      (oldVal, newVal) => {
+        if (
+          oldVal.currentSubtitleStreamIndex !==
+            newVal.currentSubtitleStreamIndex &&
+          (oldVal.currentSubtitleTrack?.DeliveryMethod ===
+            SubtitleDeliveryMethod.Encode ||
+            newVal.currentSubtitleTrack?.DeliveryMethod ===
+              SubtitleDeliveryMethod.Encode)
+        ) {
+          this.fetchCurrentMediaSource();
```

Comment:
Add comment to specify that this case is when you come from or go to a situation where the sub is burnt in so that a new stream is needed